pkg/providers/nutanix/template.go

```diff
@@ -79,7 +83,15 @@ func (ntb *TemplateBuilder) GenerateCAPISpecWorkers(clusterSpec *cluster.Spec, w
 }
 
 func (ntb *TemplateBuilder) GenerateCAPISpecSecret(clusterSpec *cluster.Spec, buildOptions ...providers.BuildMapOption) (content []byte, err error) {
-	values := buildTemplateMapSecret(clusterSpec, ntb.creds)
+	encodedCreds, _ := json.Marshal(ntb.creds)
```

Comment:
Let's not ignore the error here
```suggestion
	encodedCreds, err := json.Marshal(ntb.creds)
	if err != nil {
		return nil, fmt.Errorf("marshaling Nutanix credentials: %v", err)
	}
```
Same on line 93